de-lib/src/main/java/org/iplantc/de/admin/desktop/client/communities/presenter/AdminCommunitiesPresenterImpl.java

```diff
@@ -315,43 +311,41 @@ public Group getSelectedCommunity() {
     public void onCommunitySelectionChanged(CommunitySelectionChanged event) {
         Group community = event.getCommunity();
         AdminAppsGridView communityApps = communityGridPresenter.getView();
-        communityApps.mask(appearance.loadingMask());
+        communityApps.setLoading(true);
 
-        serviceFacade.getCommunityApps(community, new AppsCallback<List<App>>() {
+        serviceFacade.getCommunityApps(community, new AppsCallback<Splittable>() {
 
             @Override
             public void onFailure(Integer statusCode, Throwable exception) {
-                communityApps.unmask();
+                communityApps.setLoading(false);
                 ErrorHandler.post(exception);
             }
 
             @Override
-            public void onSuccess(List<App> result) {
-                communityApps.clearAndAdd(result);
-                communityApps.unmask();
+            public void onSuccess(Splittable result) {
+                communityGridPresenter.setApps(result);
```

Comment:
Actually no. But I understand why this can get confusing. The loading mask is removed in https://github.com/sri2k1us/ui/blob/89b11e930f11a1beca700d7cc48713c6e8bb528a/de-lib/src/main/java/org/iplantc/de/admin/apps/client/views/grid/AdminAppsGridImpl.java#L58

The reason for doing this way is when you call more than 1 ` CyVerseReactComponents.render` in quick succession, you will get an error saying something like `Already rendering`. So I am batch updating props here and calling ` CyVerseReactComponents.render` once. So whenever `setApps` is called in `AdminAppsGridImpl` it also sets loading to `false`.

Reply:
That brings another thing I was confused about but forgot to mention: I see a `setApps(Splittable apps, boolean loading)` method was added at https://github.com/cyverse-de/ui/pull/472/files#diff-eacc955aa276db8e0dc43aa90dab7d92R58, but I don't see a `setApps(Splittable apps)` method, which all these `setApps` method calls are using.

Reply:
Right..the method `setApps(splittable)` are in presenter which caches the list for updates and deletes. This method will in turn call `setApps(splittable,boolean)`. See https://github.com/sri2k1us/ui/blob/8da0aacc3e2ff9485502555fa8e4fcf8003a93b8/de-lib/src/main/java/org/iplantc/de/admin/apps/client/presenter/grid/AdminAppsGridPresenterImpl.java#L194

Reply:
Oh, oops. I forgot I was looking at the view and not the presenter.
Thanks for the clarification! 👍

Reply:
No problem. I wish I could simplify this code further but then I already took so much time. Also in between QA Env was down and that further delayed it. Eventually I had to focus more on Sonora.